tests/e2e/restart_test.go

```diff
@@ -20,97 +18,141 @@
 package e2e
 
 import (
-	"log"
-	"os"
-
 	. "gopkg.in/check.v1"
 
 	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
 )
 
-type PluginSuite struct {
+type RestartTestData struct {
+	config        *inputparams.TestConfig
 	volumeName    string
-	hostIP        string
 	containerName string
-	esxIP         string
 }
 
-func (s *PluginSuite) SetUpTest(c *C) {
-	s.hostIP = os.Getenv("VM2")
-	s.esxIP = os.Getenv("ESX")
+func (s *RestartTestData) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping docker/plugin restart tests")
+	}
+
 	s.volumeName = inputparams.GetUniqueVolumeName("restart_test")
 	s.containerName = inputparams.GetUniqueContainerName("restart_test")
 
+	// ensure there are no remanants from an earlier run
+	out, err := dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
+
 	// Create a volume
-	out, err := dockercli.CreateVolume(s.hostIP, s.volumeName)
+	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 }
 
-func (s *PluginSuite) TearDownTest(c *C) {
-	out, err := dockercli.RemoveContainer(s.hostIP, s.containerName)
-	c.Assert(err, IsNil, Commentf(out))
-	out, err = dockercli.DeleteVolume(s.hostIP, s.volumeName)
+func (s *RestartTestData) TearDownSuite(c *C) {
+	out, err := dockercli.DeleteVolume(s.config.DockerHosts[1], s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 }
 
-var _ = Suite(&PluginSuite{})
+var _ = Suite(&RestartTestData{})
 
-// Test Stale mount
-// 1. Attach a volume to a container
+// TestVolumeDetached - verifies a volume is detached when the docker
+// daemon is stopped.
+// 1. Attach a volume to a container on a host
 // 2. Verify attached status
 // 3. Restart docker
 // 4. Verify detached status. Volume should be detached (within the timeout)
-func (s *PluginSuite) TestVolumeStaleMount(c *C) {
-	log.Printf("START: restart_test.TestVolumeStaleMount")
-
-	out, err := dockercli.AttachVolume(s.hostIP, s.volumeName, s.containerName)
+// 5. Verify a container can be started to use the same volume on another host
```

Comment:
The test verifies that the volume is usable.

Not sure these tests were reviewed earlier but as I'm writing these tests, I'm also reviewing to see what makes the test correct and complete. For example, verifying detached state (original code) isn't right, the KV may say its detached but the volume may be stuck in a VM.

The correct verification is "can the volume be used again" which is by running a container and can it be used across docker host VMs. If a volume is stuck in a VM, then attach will work for it. So the correct test would be:

1. Run a container,
2. Restart docker
3. Assert the volume is detached (its not a sure test of correctness of volume state).
4. Attach volume to a different docker host VM - this ensures that the volume is truly detached.
5. Restart the docker on the VM in (4)
6. Verify again that on the first docker host VM we are able to re-attach the volume  - bonus verification that the original docker host is able to attach the VM.